src/dsp.py

```
@@ -0,0 +1,64 @@
+ここには PROTOCOL.md の定数（150/450/700ms 等）は入らない:
+それらは「パルス長 → 短/長/プリアンブル」を判定する decode.py 側の仕事であり、
+DSP はあくまで「閾値以上の ON 区間を (start_ms, duration_ms) として切り出す」までを担う。
+帯域(lo/hi)・閾値(threshold)はチャンネル固有パラメータなので、ハードコードせず引数で受ける
+（CLAUDE.md のアーキテクチャ方針）。
+"""
+import numpy as np
+from scipy import signal as sp_signal
+
+
+def bandpass(signal, fs, lo, hi, order=4):
+    """[lo, hi] Hz のバターワース・バンドパス（ゼロ位相）。
```

Ruff (0.15.15)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)